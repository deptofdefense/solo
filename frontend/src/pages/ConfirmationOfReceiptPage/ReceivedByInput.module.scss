.receivedByInput {
  margin-top: 0 !important;
}

.btn {
  margin-left: 8px;
  width: 100px !important;
  flex-shrink: 0;
}