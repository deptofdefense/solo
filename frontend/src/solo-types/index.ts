import { SortingRule, Filters } from "react-table";
export {
  default as createFakeApiDocs,
  defaultApiDoc,
  defaultApiResponse
} from "./apiDoc";
export * from "./warehouseUsers";
export { default as createFakeDocs, defaultDoc } from "./doc";

export interface Part {
  id: number;
  nsn: string;
  nomen: string;
  uom: string;
  price?: number;
  sac?: number;
  serial_control_flag?: string;
  lot_control_flag?: string;
  recoverability_code?: string;
  shelf_life_code?: number;
  controlled_inv_item_code?: string;
}

export interface Dic {
  id: number;
  code: string;
  desc: string;
}

export interface Status {
  id: number;
  dic: Dic;
  status_date: string;
  key_and_transmit_date?: string | null;
  esd?: string | null;
  received_qty: number | null;
  projected_qty: number;
  document: number;
  user?: number | null;
}

export interface AddressType {
  id: number;
  type: string;
  desc?: string | null;
}

export interface Address {
  id: number;
  address_type: AddressType;
  name: string;
  ric: string;
  addy1: string;
  addy2: string;
  addy3: string;
  city: string;
  state: string;
  zip: string;
  country: string;
  document: number[];
}

export interface ServiceRequest {
  id: number;
  service_request: string;
}

export interface Locator {
  id: number;
  code: string;
  desc?: string;
  subinventorys?: number;
}

export interface Subinventory {
  id: number;
  locators: Locator[];
  code: string;
  desc?: string;
  suppadd?: number;
}

export type LocatorMap = Record<string, Locator[]>;

export interface SuppAdd {
  id: number;
  code: string;
  subinventorys: Subinventory[];
}

interface BaseDocument {
  id: number;
  sdn: string;
  part: Part;
  statuses: Status[];
}

export interface ApiDocument extends BaseDocument {
  service_request: ServiceRequest;
  addresses: Address[];
  suppadd: SuppAdd;
}

export interface Document extends BaseDocument {
  serviceRequest: ServiceRequest;
  shipper?: Address;
  receiver?: Address;
  loadingStatus: LoadingStatus;
  subinventorys: Omit<Subinventory, "locators">[];
  locatorsBySubinventory: LocatorMap;
  commodityName: string;
  mostRecentStatusIdx: number;
  enteredReceivedQty: number;
  enteredReceivedBy: string;
  enteredSubinventoryCode: string;
  enteredLocatorCode: string;
}

export interface LoadingStatus {
  loading: boolean;
  error?: boolean;
  message?: string;
}

export interface Query<T extends object> {
  sort: SortingRule<T>[];
  page: number;
  filters: Filters<T>;
}

export interface PaginatedApiResponse<T> {
  results: T;
  count: number;
  next: number;
  previous: number;
}

export interface WarehouseUser extends LoadingStatus {
  userId: number;
  username: string;
  canD6T: boolean;
  canCOR: boolean;
  aac: string;
  hasModified?: boolean;
}
