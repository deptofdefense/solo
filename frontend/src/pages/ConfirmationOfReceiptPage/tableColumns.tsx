import React from "react";
import { Column } from "react-table";
import { Document } from "solo-types";
import { Checkbox } from "solo-uswds";
import ReceivedByInputCell from "./SubmitCORCell";

export interface DocumentWithReceivedBy extends Document {
  receivedBy?: string;
  submitting?: boolean;
  error?: string | null;
}

interface CreateOptions {
  onSubmitCOR: (sdn: string, receivedBy: string) => void;
}

type CreateColumns = (opts: CreateOptions) => Column<DocumentWithReceivedBy>[];

const createColumns: CreateColumns = ({ onSubmitCOR }) => [
  {
    id: "select",
    disableSortBy: true,
    Header: ({ getToggleAllRowsSelectedProps }) => {
      const {
        checked,
        onChange,
        indeterminate,
        ...rest
      } = getToggleAllRowsSelectedProps();
      return (
        <div className="padding-left-2">
          <Checkbox checked={checked} onChange={onChange} {...rest} />
        </div>
      );
    },
    Cell: ({ row }) => {
      const {
        checked,
        onChange,
        indeterminate,
        ...rest
      } = row.getToggleRowSelectedProps();
      return <Checkbox checked={checked} onChange={onChange} {...rest} />;
    }
  },
  {
    Header: "SDN",
    accessor: "sdn"
  },
  {
    Header: "NIIN",
    accessor: "part.nsn",
    id: "nsn"
  },
  {
    Header: "Quantity",
    id: "quantity",
    accessor: ({ statuses }) => statuses[statuses.length - 1]?.received_qty
  },
  {
    Header: "Commodity",
    id: "commodity",
    accessor: "suppadd.desc"
  },
  {
    Header: "Submit COR",
    id: "submitCOR",
    accessor: "receivedBy",
    disableSortBy: true,
    Cell: ({ ...args }) => (
      <ReceivedByInputCell {...args} onSubmitCOR={onSubmitCOR} />
    )
  }
];

export default createColumns;
