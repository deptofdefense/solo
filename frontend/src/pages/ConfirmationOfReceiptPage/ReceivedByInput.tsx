import React, { useState, useEffect } from "react";
import classNames from "classnames";
import { Row, Cell } from "react-table";
import { Button } from "solo-uswds";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSpinner } from "@fortawesome/free-solid-svg-icons";
import { DocumentWithReceivedBy } from "./tableColumns";
import classes from "./ReceivedByInput.module.scss";

interface ReceivedByInputCellProps {
  onSubmitCOR: (sdn: string, receivedBy: string) => void;
  cell: Cell<DocumentWithReceivedBy>;
  row: Row<DocumentWithReceivedBy>;
}

const ReceivedByInputCell: React.FC<ReceivedByInputCellProps> = ({
  onSubmitCOR,
  row: {
    original: { sdn, submitting, error }
  }
}) => {
  const [receivedBy, setReceivedBy] = useState("");
  const onSubmit: React.FormEventHandler<HTMLFormElement> = e => {
    e.preventDefault();
    onSubmitCOR(sdn, receivedBy);
  };

  useEffect(() => {
    setReceivedBy("");
  }, [sdn]);

  return (
    <form
      onSubmit={onSubmit}
      className="display-flex flex-row flex-justify-start flex-align-center"
    >
      <input
        className={classNames("usa-input", classes.receivedByInput)}
        placeholder="Rank Lastname, Firstname MI"
        disabled={submitting}
        value={receivedBy}
        onChange={e => setReceivedBy(e.currentTarget.value)}
      />
      <Button
        type="submit"
        disabled={submitting || !receivedBy}
        color={error ? "secondary" : undefined}
        className={classes.btn}
      >
        {submitting ? <FontAwesomeIcon icon={faSpinner} spin /> : "Submit"}
      </Button>
    </form>
  );
};

export default ReceivedByInputCell;
