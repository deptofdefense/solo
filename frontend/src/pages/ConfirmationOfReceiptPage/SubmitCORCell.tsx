import React, { useState, useEffect } from "react";
import { Row, Cell } from "react-table";
import { DocumentWithReceivedBy } from "./tableColumns";

import { CORInputForm } from "components";

interface ReceivedByInputCellProps {
  onSubmitCOR: (sdn: string, receivedBy: string) => void;
  cell: Cell<DocumentWithReceivedBy>;
  row: Row<DocumentWithReceivedBy>;
}

const ReceivedByInputCell: React.FC<ReceivedByInputCellProps> = ({
  onSubmitCOR,
  row: {
    original: { sdn, submitting, error }
  }
}) => {
  const [receivedBy, setReceivedBy] = useState("");

  const onSubmit = () => {
    onSubmitCOR(sdn, receivedBy);
  };

  useEffect(() => {
    setReceivedBy("");
  }, [sdn]);

  return (
    <CORInputForm
      value={receivedBy}
      onReceivedByChange={setReceivedBy}
      onSubmitCOR={onSubmit}
      error={error}
      loading={submitting}
    />
  );
};

export default ReceivedByInputCell;
