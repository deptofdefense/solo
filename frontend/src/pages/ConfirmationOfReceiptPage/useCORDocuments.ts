import { useCallback, useState } from "react";
import { useDocumentSet } from "hooks";
import { Query, Document } from "solo-types";
import { DocumentWithReceivedBy } from "./tableColumns";
import { useAuthContext } from "context";
import { sleep } from "test-utils";

const useCORDocuments = () => {
  const { apiCall } = useAuthContext();
  const [bulkSubmitStatus, setBulkSubmitStatus] = useState<{
    submitting: boolean;
    error?: string;
  }>({
    submitting: false
  });
  const {
    setDocs,
    docs,
    updateDocuments,
    modifyDocument,
    removeDocument,
    ...rest
  } = useDocumentSet<DocumentWithReceivedBy>();

  const updateDocumentsWithD6TStatus = useCallback(
    (query: Query<Document>) =>
      // always filter for documents that are ready for COR
      updateDocuments({
        ...query,
        filters: [
          ...query.filters,
          {
            id: "status",
            value: "D6T"
          }
        ]
      }),
    [updateDocuments]
  );

  const submitCOR = useCallback(
    async (sdn: string, receivedBy: string) => {
      modifyDocument(sdn, {
        submitting: true
      });
      // simulate up to 1.5 seconds of load time
      await sleep(Math.floor(Math.random() * 1500));
      try {
        // submit single document and remove from current list
        await apiCall("/document/cor", {
          method: "POST",
          body: JSON.stringify({
            sdn,
            receivedBy,
            status: "COR"
          })
        });
        removeDocument(sdn);
      } catch (e) {
        // api or network error
        modifyDocument(sdn, {
          submitting: false,
          error: e.message || "Something went wrong"
        });
      }
    },
    [modifyDocument, removeDocument, apiCall]
  );

  const submitBulkCOR = useCallback(
    async (sdns: string[], receivedBy: string) => {
      setBulkSubmitStatus({
        submitting: true
      });
      // simulate up to 1.5 seconds of load time
      await sleep(Math.floor(Math.random() * 1500));
      try {
        // submit bulk documents and remove from list
        await apiCall("/document/cor/bulk", {
          method: "POST",
          body: JSON.stringify(
            sdns.map(sdn => ({
              sdn,
              receivedBy,
              status: "COR"
            }))
          )
        });
        // filter all submitted sdns
        setDocs(prevDocs =>
          prevDocs.filter(({ sdn }) => sdns.indexOf(sdn) < 0)
        );
        setBulkSubmitStatus({
          submitting: false
        });
      } catch (e) {
        setBulkSubmitStatus({
          submitting: false,
          error: e.message || "Something went wrong"
        });
      }
    },
    [setBulkSubmitStatus, setDocs, apiCall]
  );

  const resetBulkSubmitStatus = useCallback(() => {
    setBulkSubmitStatus({
      submitting: false
    });
  }, [setBulkSubmitStatus]);

  return {
    setDocs,
    docs,
    updateDocuments: updateDocumentsWithD6TStatus,
    submitCOR,
    submitBulkCOR,
    bulkSubmitStatus,
    resetBulkSubmitStatus,
    ...rest
  };
};

export default useCORDocuments;
