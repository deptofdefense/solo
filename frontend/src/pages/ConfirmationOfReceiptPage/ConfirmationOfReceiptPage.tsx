import React, { useMemo, useState, useCallback } from "react";
import { TableInstance } from "react-table";
import {
  Title,
  Table,
  SelectFilterControls,
  Paginator,
  CORInputForm
} from "components";
import { Document } from "solo-types";
import useCORDocuments from "./useCORDocuments";
import createColumns from "./tableColumns";

const filterable = [
  { name: "SDN", value: "sdn" },
  { name: "Nomenclature", value: "nomen" },
  { name: "Commodity", value: "commod" }
];

const ConfirmationOfReceiptPage: React.FC = () => {
  const [bulkReceivedBy, setBulkReceivedBy] = useState("");
  const {
    docs,
    updateDocuments,
    pageCount,
    submitCOR,
    submitBulkCOR,
    bulkSubmitStatus,
    resetBulkSubmitStatus
  } = useCORDocuments();
  const columns = useMemo(
    () =>
      createColumns({
        onSubmitCOR: submitCOR
      }),
    [submitCOR]
  );

  const onSelectedRowsChange = useCallback(
    ({ toggleHideColumn, selectedFlatRows }: TableInstance<Document>) => {
      // show individual row submit forms only when no rows are selected
      resetBulkSubmitStatus();
      toggleHideColumn("submitCOR", selectedFlatRows.length > 0);
    },
    [resetBulkSubmitStatus]
  );

  const renderPagination = (table: TableInstance<Document>) => (
    <>{table.selectedFlatRows.length === 0 && <Paginator table={table} />}</>
  );

  const renderFilterControls = (table: TableInstance<Document>) => {
    const { setGlobalFilter, selectedFlatRows } = table;
    return selectedFlatRows.length > 0 ? (
      <CORInputForm
        value={bulkReceivedBy}
        onReceivedByChange={setBulkReceivedBy}
        onSubmitCOR={() =>
          submitBulkCOR(
            selectedFlatRows.map(({ original }) => original.sdn),
            bulkReceivedBy
          )
        }
        actionText={`Submit ${selectedFlatRows.length} Cors`}
        className="margin-left-2 padding-y-2 flex-justify-center"
        {...bulkSubmitStatus}
      />
    ) : (
      <SelectFilterControls options={filterable} onSubmit={setGlobalFilter} />
    );
  };

  return (
    <div className="tablet:margin-x-8 overflow-x-auto">
      <Title>Confirmation of Receipt</Title>
      <Table<Document>
        columns={columns}
        data={docs}
        onSelectedRowsChange={onSelectedRowsChange}
        renderFilterControls={renderFilterControls}
        renderPagination={renderPagination}
        pageCount={pageCount}
        fetchData={updateDocuments}
      />
    </div>
  );
};

export default ConfirmationOfReceiptPage;
