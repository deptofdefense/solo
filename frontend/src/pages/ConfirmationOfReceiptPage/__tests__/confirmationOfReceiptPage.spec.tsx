import React from "react";
import ConfirmationOfReceiptPage from "../ConfirmationOfReceiptPage";
import { render, wait, fireEvent } from "test-utils";
import { defaultApiResponse } from "solo-types";

describe("ConfirmationOfReceiptPage Component", () => {
  const fetchMock = jest.fn();
  const defaultDoc = defaultApiResponse.results[0];

  beforeEach(() => {
    fetchMock.mockResolvedValue(defaultApiResponse);
  });

  afterEach(() => {
    fetchMock.mockReset();
  });

  it("matches snapshot", async () => {
    const { asFragment, queryByText } = render(<ConfirmationOfReceiptPage />, {
      authContext: {
        apiCall: fetchMock
      }
    });
    await wait(() => {
      // wait for some data to render before snapshot test
      expect(fetchMock).toHaveBeenCalled();
      expect(queryByText(defaultDoc.sdn)).toBeInTheDocument();
      expect(queryByText(defaultDoc.suppadd.desc)).toBeInTheDocument();
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it("adds a filter for a status of d6t on initial load", async () => {
    render(<ConfirmationOfReceiptPage />, {
      authContext: {
        apiCall: fetchMock
      }
    });
    await wait(() => {
      expect(fetchMock).toHaveBeenCalled();
      expect(fetchMock.mock.calls[0][0]).toEqual(
        "/document/?status=D6T&exclude_status=COR"
      );
      expect(fetchMock.mock.calls[0][1]).toMatchObject({
        method: "GET"
      });
    });
  });

  it("adds d6t status filter to any filters submitted by user", async () => {
    const { getByPlaceholderText, container } = render(
      <ConfirmationOfReceiptPage />,
      {
        authContext: {
          apiCall: fetchMock
        }
      }
    );
    const filterInp = getByPlaceholderText("Search");
    const filterSubmit = container.querySelector(
      "button[type='submit']"
    ) as Element;
    fireEvent.change(filterInp, {
      target: { value: "usersdnfilter" }
    });
    await wait(() => {
      expect(filterInp).toHaveValue("usersdnfilter");
    });
    fireEvent.click(filterSubmit);
    await wait(() => {
      // first call was on render
      expect(fetchMock).toHaveBeenCalledTimes(2);
      expect(fetchMock.mock.calls[1][0]).toEqual(
        "/document/?sdn=usersdnfilter&status=D6T&exclude_status=COR"
      );
    });
  });

  it("adds d6t status filter when paginating due-outs", async () => {
    const { getByText, queryByText } = render(<ConfirmationOfReceiptPage />, {
      authContext: {
        apiCall: fetchMock
      }
    });
    await wait(() => {
      expect(queryByText(defaultDoc.sdn)).toBeInTheDocument();
    });
    const pageTwo = getByText("2");
    fireEvent.click(pageTwo);
    await wait(() => {
      // first call was on render
      expect(fetchMock).toHaveBeenCalledTimes(2);
      expect(fetchMock.mock.calls[1][0]).toEqual(
        "/document/?page=2&status=D6T&exclude_status=COR"
      );
    });
  });
});
