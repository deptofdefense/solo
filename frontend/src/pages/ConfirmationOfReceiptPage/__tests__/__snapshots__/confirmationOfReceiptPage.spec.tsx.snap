// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationOfReceiptPage Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="tablet:margin-x-8 overflow-x-auto"
  >
    <h1
      class="usa-prose"
    >
      Confirmation of Receipt
    </h1>
    <form
      class="grid-row flex-row flex-justify-center flex-align-center"
    >
      <div
        class="grid-col-full tablet:grid-col-2 margin-x-2"
      >
        <select
          class="usa-select"
          placeholder="Field"
        >
          <option
            value="sdn"
          >
            SDN
          </option>
          <option
            value="nomen"
          >
            Nomenclature
          </option>
          <option
            value="commodity"
          >
            Commodity
          </option>
        </select>
      </div>
      <div
        class="grid-col-10 tablet:grid-col-3"
      >
        <input
          class="usa-input select"
          name="input-type-text"
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
      <div
        class="grid-col-auto"
      >
        <button
          class="usa-button margin-top-1 square"
          type="submit"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-search fa-w-16 "
            data-icon="search"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </form>
    <table
      class="usa-table grid-col-12 usa-table--borderless"
      role="table"
    >
      <thead>
        <tr
          role="row"
        >
          <th
            colspan="1"
            role="columnheader"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              <div
                class="padding-left-2"
              >
                <input
                  class="usa-checkbox__input inp"
                  style="cursor: pointer;"
                  title="Toggle All Rows Selected"
                  type="checkbox"
                />
              </div>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              SDN
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              NIIN
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              Quantity
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              Commodity
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              Submit COR
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            class="td"
            role="cell"
          >
            <input
              class="usa-checkbox__input inp"
              style="cursor: pointer;"
              title="Toggle Row Selected"
              type="checkbox"
            />
          </td>
          <td
            class="td"
            role="cell"
          >
            M3030012341234
          </td>
          <td
            class="td"
            role="cell"
          >
            5430015061999
          </td>
          <td
            class="td"
            role="cell"
          />
          <td
            class="td"
            role="cell"
          >
            Motor Transportation
          </td>
          <td
            class="td"
            role="cell"
          >
            <div
              class="display-flex flex-column"
            >
              <form
                class="display-flex flex-row flex-align-center"
              >
                <input
                  class="usa-input receivedByInput"
                  placeholder="Rank Lastname, Firstname MI"
                  value=""
                />
                <button
                  class="usa-button btn"
                  disabled=""
                  type="submit"
                >
                  Submit
                </button>
              </form>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="grid-row flex-justify-center margin-y-4"
    >
      <button
        class="usa-button pageLink disabled usa-button--unstyled"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-left fa-w-10 fa-sm "
          data-icon="chevron-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-left fa-w-10 fa-sm margin-right-1"
          data-icon="chevron-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
            fill="currentColor"
          />
        </svg>
        First
      </button>
      <button
        class="usa-button pageLink disabled usa-button--unstyled"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-left fa-w-10 fa-sm margin-right-1"
          data-icon="chevron-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
            fill="currentColor"
          />
        </svg>
        Previous
      </button>
      <button
        class="usa-button pageLink usa-button--outline"
      >
        1
      </button>
      <button
        class="usa-button pageLink usa-button--unstyled"
      >
        2
      </button>
      <button
        class="usa-button pageLink usa-button--unstyled"
      >
        3
      </button>
      <button
        class="usa-button pageLink usa-button--unstyled"
      >
        Next
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-right fa-w-10 fa-sm margin-left-1"
          data-icon="chevron-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="usa-button pageLink usa-button--unstyled"
      >
        Last
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-right fa-w-10 fa-sm margin-left-1"
          data-icon="chevron-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-right fa-w-10 fa-sm "
          data-icon="chevron-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
